File: dsa learning series/PRIGAME.py
MAX = 1000001
prefix =[0]*(MAX + 1)
counter =[0]*(MAX + 1)

  
def buildPrefix(): 
    prime = [1]*(MAX + 1) 
    p = 2
    while(p * p <= MAX):  
        if (prime[p] == 1): 
            i = p * 2
            while(i <= MAX): 
                prime[i] = 0
                i += p 
        p+=1
    for i in range(2, MAX+1):
        counter[i] = counter[i-1]
        if prime[i]==1:
            counter[i]+=1

buildPrefix()
for _ in(range(int(input()))):
    x, y = map(int, input().split())
    dic = {0:'Chef', 1:'Divyam '}
    if y==1:
        if x<=2:
            print(dic[0])
        else:
            print(dic[1])
    else:
        ans = counter[x]
        if ans<=y:
            print(dic[0])
        else:
            print(dic[1])

File: dsa learning series/prigame.cpp
#include<bits/stdc++.h>
using namespace std;
bool isPrime(long long int n) 
{
    if (n <= 1) 
        return false; 
    if (n <= 3) 
        return true; 
    if (n % 2 == 0 || n % 3 == 0) 
        return false; 
    long long int i;
    for (i = 5; i * i <= n; i = i + 6) 
        if (n % i == 0 || n % (i + 2) == 0) 
            return false; 
    return true; 
}
int main(){
    ios_base::sync_with_stdio(false);
    cin.tie(NULL);
    long long int t, i, counter=0;
    vector <long long int> prime(1000001);
    cin>>t;
    prime[0]=prime[1]=0;
    for(i=2;i<1000001;i++){
        if(isPrime(i))
            counter++;
        prime[i]=counter;
    }
    while(t--){
        long long int x, y;
        cin>>x>>y;
        prime[x]<=y?printf("Chef\n"):printf("Divyam\n");
    }
    return 0;
}